src/agoradatatools/process.py

```diff
@@ -18,57 +18,53 @@
 
 # TODO refactor to avoid so many if's - maybe some sort of mapping to callables
 def apply_custom_transformations(
-    datasets: dict, dataset_name: str, dataset_obj: dict
+    datasets: dict,
+    dataset_name: str,
+    dataset_obj: dict,
 ) -> Union[DataFrame, dict, None]:
-    if not isinstance(datasets, dict) or not isinstance(dataset_name, str):
+    """Apply custom transformations to the dataset based on the provided function names and parameters.
```

Comment:
the overall logic is basically: 
Retrieve the transformation function from the configuration using the dataset name, then extract any custom parameters defined in the config (e.g., for distribution_data). Next, construct a complete dictionary of parameters by combining inputs such as `dataset_name`, `df` and config-defined values. Lastly, pass this dictionary to the transformation function.